src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java

```diff
@@ -63,18 +63,24 @@ public void run() {
 
         // Upload the letters in batches.
         // With each batch we mark them Uploaded so they no longer appear in the query.
-        List<Letter> letters;
         int counter = 0;
+        int uploaded;
 
         do {
-            letters = repo.findFirst10ByStatus(LetterStatus.Created);
-            letters
-                .forEach(letter -> {
-                    uploadToFtp(letter);
-                    markAsUploaded(letter);
-                });
-            counter += letters.size();
-        } while (!letters.isEmpty());
+            uploaded = counter;
+            // applying single connection per single loop run.
+            // should be a significant speed improvement already reported
+            counter += ftp.runWith(sftpClient ->
```

Comment:
Since our goal here is to address the problem of opening the connection each time we want to upload a file, why not solve it entirely and only open it once every run of the task?

Reply:
this might introduce sudden connection drops especially after downtime window when theoretically letters pile up. or whenever there are loads of requests done during some point of the day. the approach is start small and review. the suggestion of yours is a good candidate for next pr along with _keep alive_ profile for ssh client

Reply:
When we open the connection once per run, then even if that connection is broken, the files will get uploaded within a new connection, in the next run, which will be in 30 seconds. Given that opening a connection once per job run seems to be about as much of a change as this, I don't see why we shouldn't do it in the first go.

Reply:
To be clear, I'm not blocking the PR with this comment

Reply:
Another improvement, not necessarily in this PR, but still something we should do, would be not opening a connection if there's nothing to send.